src/Storages/System/StorageSystemKafkaConsumers.cpp

```diff
@@ -152,6 +177,78 @@ void StorageSystemKafkaConsumers::fillData(MutableColumns & res_columns, Context
             last_used.insert(static_cast<Decimal64>(consumer_stat.last_used_usec));
 
             rdkafka_stat.insertData(consumer_stat.rdkafka_stat.data(), consumer_stat.rdkafka_stat.size());
+
+            auto add_missing_dependency = [&](const std::vector<String>& route_)
```

Comment:
I think in general the code is really good, however would it be possible to calculate the routes only once and reuse it for all consumers in the same table? This shouldn't be a problem, but I guess if somebody has a lot of consumers and a lot of MVs at the same table, this way we can return different missing/ready dependencies for consumers from the same table. Also, reusing the calculated values are less wasteful .It is not a computationally heavy operation, but let's not traverse the same dependencies again and again.